src/battle-log.ts

```diff
@@ -805,7 +805,16 @@ class BattleLog {
 				const color = this.usernameColor(toID(getAttrib('name')));
 				const style = getAttrib('style');
 				setAttrib('style', `${style};color:${color}`);
-			} else if (tagName === 'youtube') {
+			} else if (tagName === 'spotify') {
+				// <iframe src="https://open.spotify.com/embed/track/6aSYnCIwcLpnDXngGKAEzZ" width="300" height="380" frameborder="0" allowtransparency="true" allow="encrypted-media"></iframe>
+				const src = getAttrib('src');
+				const songId = /(?:\?v=|\/track\/)([A-Za-z0-9]+)/.exec(src)[1];
```

Comment:
```suggestion
				const songId = /(?:\?v=|\/track\/)([A-Za-z0-9]+)/.exec(src)?.[1];
```

Reply:
There should probably be a check to not return a tag if `songId` is undefined. I'm not familar with the tag parser but https://open.spotify.com/embed/track/undefined doesn't seem like a very valid URL :p